pkg/conversion/mapi2capi/powervs_test.go

```diff
@@ -0,0 +1,140 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+var _ = Describe("mapi2capi PowerVS", func() {
+
+	powerVSProviderSpec := getPowerVSProviderSpec()
+
+	powerVSRawExt, err := rawExtensionFromPowerVSProviderSpec(powerVSProviderSpec)
+	Expect(err).ToNot(HaveOccurred())
+
+	powerVSMAPIMachine := &machinev1beta1.Machine{
+		Spec: machinev1beta1.MachineSpec{
+			ProviderSpec: machinev1beta1.ProviderSpec{
+				Value: powerVSRawExt,
+			},
+			ProviderID: ptr.To("test-123"),
+		},
+	}
+
+	powerVSMAPIMachineSet := &machinev1beta1.MachineSet{
+		Spec: machinev1beta1.MachineSetSpec{
+			Selector: metav1.LabelSelector{},
+			Template: machinev1beta1.MachineTemplateSpec{
+				Spec: machinev1beta1.MachineSpec{
+					ProviderSpec: machinev1beta1.ProviderSpec{
+						Value: powerVSRawExt,
+					},
+					ProviderID: ptr.To("test-123"),
+				},
+			},
+		},
+	}
+
+	infra := configv1.Infrastructure{
+		Spec: configv1.InfrastructureSpec{},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName: "sample-cluster-name",
+		},
+	}
+
+	// We need to add error handling for the providerSpec issues before we can enable the following tests.
+	It("should be able to convert a MAPI Power VS Machine to a CAPI Machine", func() {
+		// Convert a MAPI Machine to a CAPI Core Machine + a CAPI InfraMachine.
+		capiMachine, capiInfraMachine, warns, err :=
+			FromPowerVSMachineAndInfra(powerVSMAPIMachine, &infra).ToMachineAndInfrastructureMachine()
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert providerSpec to MachineTemplateSpec")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting providerSpec to MachineTemplateSpec")
+		Expect(capiMachine).To(Not(BeNil()), "should not have a nil CAPI Machine")
+		Expect(capiInfraMachine).To(Not(BeNil()), "should not have a nil CAPI InfrastructureMachine")
+	})
+
+	It("should be able to convert a MAPI Power VS MachineSet to a CAPI MachineSet", func() {
+		// Convert a MAPI MachineSet to a CAPI Core MachineSet + a CAPI InfraMachineTemplateSpec.
+		capiMachineSet, capiInfraMachineTemplate, warns, err :=
+			FromPowerVSMachineSetAndInfra(powerVSMAPIMachineSet, &infra).ToMachineSetAndMachineTemplate()
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert MAPI MachineSet to CAPI MachineSet")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting MAPI MachineSet to CAPI MachineSet")
+		Expect(capiMachineSet).To(Not(BeNil()), "should not have a nil CAPI MachineSet")
+		Expect(capiInfraMachineTemplate).To(Not(BeNil()), "should not have a nil CAPI MachineTemplate")
+	})
+
+})
+
+// getPowerVSProviderSpec builds and returns PowerVSProviderConfig.
+func getPowerVSProviderSpec() *mapiv1.PowerVSMachineProviderConfig {
+	// TODO: May be we can add this in machine builder?
```

Comment:
Would be good to be consistent with how we've done this, the machine builder pattern is pretty nice for writing these kinds of tests

Reply:
I will follow up on this to add support for Power VS.